mw-webapp/src/pages/authPage/AuthPage.tsx

```diff
@@ -1,17 +1,37 @@
-import {Link} from "react-router-dom";
-import {useAuth} from "src/utils/useAuth";
+import {User} from "firebase/auth";
+import {useEffect, useState} from "react";
+import {Link, useNavigate} from "react-router-dom";
+import {Button} from "src/component/button/Button";
+import {getNewUserCredentials} from "src/utils/auth/getNewUserCredentials";
+import {handleUserAuthState} from "src/utils/auth/handleUserAuthState";
+import {handleLogIn} from "src/utils/auth/handleLogIn";
+import {handleLogOut} from "src/utils/auth/handleLogOut";
 
 export const AuthPage = () => {
-  const handleGoogleSignIn = useAuth();
+  const [user, setUser] = useState<User | null>(null);
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    handleUserAuthState(setUser);
+    return () => {
+      getNewUserCredentials();
+    };
+  }, []);
 
   return (
     <>
-      <button onClick={handleGoogleSignIn}>
-        Sign in with Google
-      </button>
-      <Link to={"main"}>
-        Workflow
-      </Link>
+      {user ?
+        navigate("/main") : (
```

Comment:
create consts with paths. Think maybe not in this file

Reply:
When we need to use path consts somewhere else we will create separate file.
For now, I create this const in the same file